src/sql/statement.rs

```diff
@@ -670,15 +720,18 @@ fn handle_create_sink(scx: &StatementContext, stmt: Statement) -> Result<Plan, f
     let catalog_entry = scx.catalog.get(&from)?;
 
     let topic_name = format!(
-        "{}-{}-{}",
+        "{}-{}-{}-{}",
         topic,
         scx.catalog
             .creation_time()
             .duration_since(UNIX_EPOCH)?
             .as_secs(),
-        scx.catalog.nonce()
+        scx.catalog.nonce(),
+        scx.catalog.sink_count(),
```

Comment:
@benesch we can't naively use allocate_id here because we have a immutable ref to the catalog and thats threaded all the way up -- I figured it was best to add a new counter for what we needed rather than going through a bunch of hoops to get GlobalIds in the way we want? I can add comments / file followups to get rid of this if you'd like?

Reply:
Oh man, sorry this is turning into such a quagmire. That seems reasonable, but I'm not seeing the call to increment_sink_count, and also as currently written I believe the sink count can go backwards after a crash and restart? Maybe that's ok though? 

An alternative for your consideration: move `create_sink_kafka_topic` into coord, and call it where you do have mutable access to the catalog. I haven't tested that though. Just do whatever's easiest for now and we can clean it up later!

Reply:
Its lines 460 - 462 in src/catalog/lib.rs
```
        if let CatalogItem::Sink(_) = entry.item() {
            self.increment_sink_count();
        }
```
its ok for the sink count to change over restarts because we'll use the new start time / nonce

if we move `create_sink_kafka_topic` -- then we can't initialize the `KafkaSinkConnector` in this part of the planner either because we need a topic name :| I think for now this is easiest